pkg/library/container/mountSources.go

```diff
@@ -84,20 +88,43 @@ func handleMountSources(k8sContainer *corev1.Container, devfileContainer *dw.Con
 		Name:  devfileConstants.ProjectsSourceEnvVar,
 		Value: path.Join(sourceMapping, projectsSourcePath),
 	})
+
+	return nil
 }
 
-// getProjectSourcePath gets the path, relative to PROJECTS_ROOT, that should be used for the PROJECT_SOURCE env var
-func getProjectSourcePath(projects []dw.Project) string {
-	projectPath := ""
+// getProjectSourcePath gets the path, relative to PROJECTS_ROOT, that should be used for the PROJECT_SOURCE env var.
+// Returns an error if there was a problem retrieving the selected starter project.
+//
+// The project source path is determined based on the following priorities:
+//
+//  1. If the workspace has at least one regular project, the first one will be selected.
+//     If the first project has a clone path, it will be used, otherwise the project's name will be used as the project source path.
+//
+//  2. If the workspace has a starter project that is selected, its name will be used as the project source path.
+//
+//  3. If the workspace has at least one starter project, but none are selected,
+//     the name of the first starter project will be used as the project source path.
```

Comment:
I think this may break more than it fixes. If there are starterProjects but none selected, then `$PROJECT_SOURCE` will point to a nonexistent directory since project-clone won't clone anything.

It might be better to just take 1, 2, and 4 from this list. From my understanding, what Che will do is automatically "select" the first starterProject anyways, so this won't break their use case either.

Reply:
> I think this may break more than it fixes. If there are starterProjects but none selected, then `$PROJECT_SOURCE` will point to a nonexistent directory since project-clone won't clone anything.

Good catch, I hadn't considered that. Keeping points 1,2 and 4 as suggested.